Packages/ClientRuntime/Sources/PrimitiveTypeExtensions/AsyncSequence+Extension.swift

```diff
@@ -0,0 +1,22 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+extension AsyncSequence {
+    public func asyncCompactMap<T>(
+        _ transform: (Element) -> [T]?
+    ) async rethrows -> [T] {
+        var values = [T]()
+
+        for try await element in self {
+            if let element = transform(element) {
+                values.append(contentsOf: element)
+            }
+        }
+
+        return values
+    }
```

Comment:
I guess this is a transformer to collect an async iterator into a list? a doc comment would probably be useful if customers are going to use it